src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2249,6 +2265,30 @@ def update_azure_service_mesh_profile(self) -> ServiceMeshProfile:
             new_profile.istio.certificate_authority.plugin.cert_chain_object_name = cert_chain_object_name
             updated = True
 
+        # deal with mesh upgrade commands
+        if mesh_upgrade_command is not None:
+            requested_revision = self.raw_param.get("revision", None)
+            if mesh_upgrade_command == CONST_AZURE_SERVICE_MESH_UPGRADE_COMMAND_COMPLETE or mesh_upgrade_command == CONST_AZURE_SERVICE_MESH_UPGRADE_COMMAND_ROLLBACK:
+                if len(new_profile.istio.revisions) < 2:
+                    raise ArgumentUsageError('Azure Service Mesh upgrade is not in progress.')
+                if mesh_upgrade_command == CONST_AZURE_SERVICE_MESH_UPGRADE_COMMAND_COMPLETE:
+                    revision_to_remove = min(new_profile.istio.revisions)
```

Comment:
Will `min` and `max` here to a string comparison based on semver? There was a similar concern with the sorting at RP validations: https://msazure.visualstudio.com/DefaultCollection/CloudNativeCompute/_git/aks-rp/pullRequest/8764389#1695863750

Reply:
thanks for pointing this out, added code to validate incoming --revision param value and also added semantic sort.